datafusion/datasource-parquet/src/row_filter.rs

```diff
@@ -464,18 +464,15 @@ pub fn build_row_filter(
     // `a = 1 AND b = 2 AND c = 3` -> [`a = 1`, `b = 2`, `c = 3`]
     let predicates = split_conjunction(expr);
 
-    let file_schema = Arc::new(file_schema.clone());
```

Comment:
these are deep clones (clone the entire `Schema`) where as with this PR only the Arc is cloned (ref count incremented)

Reply:
Yup makes total sense, I was just cautious to update the signature for `build_row_filter` since it seems to be a public function.